### PR TITLE
Updated things

### PR DESCRIPTION
**Pull request by resynth1943**
I've updated metadata, added an `npmignore` and a `gitignore`, cleaned up the `package.json`, and added a programmatic API, e.g

```js
const { createPaste } = require('hastebin');
createPaste('abc').then(processPasteUrl);
```

so you can use Hastebin in node without "shelling out".

Oh, and I've updated the README and tests to match. Have I missed anything?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,16 +28,31 @@ Examples:
...
 ```
 
+---
+
+### Via the JavaScript API
+
+```js
+var hastebin = require('hastebin')
+
+hastebin.createPaste('url', {
```
Comment (rahatarmanahmed) [inline README.md]: Can you change the example text from `'url'` to something more... text like? Took me a second to figure out that was the text content of the paste.

File: test/test.js [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
 var got = require('got')
+var jsapi = require('../src/hastebin')
 
-getStdin()
+var testData = 'test'
 
-.then(function (stdin) {
-  var hastebinUrl = stdin.trim()
-  assert(hastebinUrl.match(/^https:\/\/hastebin.com\/.*/) != null, 'stout must be a hastebin url')
+function testOutput (input) {
```
Comment (rahatarmanahmed) [inline test/test.js]: Can you rename this to something more descriptive like `assertHastebinUrlMatchesTestData` or `assertHastebinUrlMatchesText`

File: src/cli.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+.then(function (result) {
+  if (!result.body || !result.body.key) {
+    throw new Error('Did not receive hastebin key.')
+  }
+  if (argv.raw) {
+    console.log(url.resolve(hasteServer, 'raw/' + result.body.key))
+  } else {
+    console.log(url.resolve(hasteServer, result.body.key))
+  }
+})
```
Comment (rahatarmanahmed) [inline src/cli.js]: Why not reuse the programmatic API?

**Review by rahatarmanahmed — COMMENTED** [review]
A really solid contribution, thanks!

**Comment by rahatarmanahmed** [comment]
Also, if you want to use es6 you're welcome to bump the engine requirement to node 6. But keeping node 4 support is okay too 👍

File: src/cli.js (hunk 3 above)
Comment (resynth1943) [reply src/cli.js]: I can't argue with that, lol

File: README.md (hunk 1 above)
Comment (resynth1943) [reply README.md]: Oh, sorry about that. I was inbetween two different things and forgot that that's the parameter for the *content*.

**Comment by resynth1943** [comment]
@rahatarmanahmed I don't see the point of bumping the minimum engine version requirement up to v6 just for fancy syntax, to be honest. Thanks, though :)

File: test/test.js (hunk 2 above)
Comment (resynth1943) [reply test/test.js]: Done

File: README.md (hunk 1 above)
Comment (resynth1943) [reply README.md]: Boom :D

File: src/cli.js (hunk 3 above)
Comment (resynth1943) [reply src/cli.js]: Consider it done :)

**Comment by resynth1943** [comment]
By the way, would you be open (pardon the pun) to me adding a flag that automatically opens the link in the browser, via a module like `open`?

e.g

```sh
echo "abc" | hastebin --browser
# or 
echo | "abc" | hastebin -b (or -B?)
```

This would mitigate having to click the link / copy it into the omnibar. It's lazy, I know :D

File: README.md (hunk 1 above)
Comment (resynth1943) [reply README.md]: Pushed :+1:

**Comment by rahatarmanahmed** [comment]
I'm not opposed to it but personally I think functionality like that is best implemented by composing other unix tools. Piping into `xclip` or `open` is free ;)

**Review by rahatarmanahmed — APPROVED** [review]

**Comment by rahatarmanahmed** [comment]
Published in v0.1.0!

**Comment by resynth1943** [comment]
Yay! :D

**Pull request merged**